normandy/recipes/tests/test_commands.py

```diff
@@ -257,3 +258,54 @@ def test_it_does_not_resign_up_to_date_actions(self, settings, mocked_autograph)
         call_command("update_action_signatures")
         a.refresh_from_db()
         assert a.signature.signature == "original signature"
+
+
+@pytest.mark.django_db
+class TestUpdateAddonUrls(object):
+    def test_it_works(self):
+        action = ActionFactory(name="opt-out-study")
+        recipe = RecipeFactory(
+            action=action,
+            arguments={"addonUrl": "https://before.example.com/extensions/addon.xpi"},
+        )
+        call_command("update_addon_urls", "after.example.com")
+
+        recipe = Recipe.objects.get(id=recipe.id)
```

Comment:
So, if you replace this line with `recipe.refresh_from_db()` the next line below fails?

Reply:
Yeah. I'm really not sure why. I don't think we are doing any caching or anything. I didn't dig too deep though.